File: leetcode/exercises/1.0.easy.contains_duplicate/contains_duplicate_1.py
# -----------------------
# Created       : 22/10/2025
# Last Edited   : 22/10/2025
# Big O         :
# Topics        : 
# Problem Id    : 217
# -----------------------

class Solution:
	def containsDuplicate(self, nums: list[int]) -> bool:
		a = set()
		for i in nums:
			if i not in a:
				a.add(i)
			else:
				return True
		return False

sol = Solution()
print()
print(sol.containsDuplicate([2, 1, 10, 12, 1]))

File: leetcode/exercises/1.0.easy.contains_duplicate/contains_duplicate_1.js
// -----------------------
// Created       : 24/10/2025
// Last Edited   : 24/10/2025
// Big O         :
// Topics        : 
// Problem Id    : 217
// -----------------------

/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    const counts = {};
    for (const n of nums){
        counts[n] = (counts[n] ?? 0) + 1;
        if (counts[n] > 1){ return true; };
    };
    return false
};

console.log(containsDuplicate([1, 2, 3, 10, 8]))

